### PR TITLE
1170: Use root navigator when pushing subroutes

### PR DESCRIPTION
**Pull request by michael-markl**
### Short description

Subroutes are pushed to the root navigator.

### Proposed changes

Subroutes are pushed to the root navigator.

When navigating to map from the detail page of a search result, we have to first pop the detail page from the navigator stack and then navigate to the map tab.

### Side effects

None

### Resolved issues

Fixes: #1170 #1288

### CODE REVIEW

File: frontend/lib/search/results_loader.dart [hunk 1]
```diff
@@ -8,6 +8,8 @@ import 'package:infinite_scroll_pagination/infinite_scroll_pagination.dart';
 
 import 'package:ehrenamtskarte/l10n/translations.g.dart';
 
+import '../home/home_page.dart';
```
Comment (f1sh1918) [inline frontend/lib/search/results_loader.dart:11]: please use package import

File: frontend/lib/store_widgets/accepting_store_summary.dart [hunk 2]
```diff
@@ -11,18 +11,20 @@ import 'package:intl/intl.dart';
 
 import 'package:ehrenamtskarte/l10n/translations.g.dart';
 
+import '../map/map_page.dart';
```
Comment (f1sh1918) [inline frontend/lib/store_widgets/accepting_store_summary.dart:14]: please use correct import

File: frontend/lib/store_widgets/detail/detail_page.dart [hunk 3]
```diff
@@ -11,11 +11,13 @@ import 'package:graphql_flutter/graphql_flutter.dart';
 
 import 'package:ehrenamtskarte/l10n/translations.g.dart';
 
+import '../../map/map_page.dart';
```
Comment (f1sh1918) [inline frontend/lib/store_widgets/detail/detail_page.dart:14]: please adjust import

File: frontend/lib/about/language_change.dart [hunk 4]
```diff
@@ -28,17 +26,12 @@ class LanguageChange extends StatelessWidget {
...
     LocaleSettings.setLocaleRaw(language);
-    settings.setLanguage(language: language);
-    Navigator.push(
-      context,
-      AppRoute(
-        builder: (context) => HomePage(),
-      ),
-    );
+    await settings.setLanguage(language: language);
+    Navigator.pop(context);
```
Comment (f1sh1918) [inline frontend/lib/about/language_change.dart:34]: this change will avoid showing the snackbar of the language change.
You could move the showSnackbar before the `settings.setLanguage`
Not sure if there is a better solution

**Review by f1sh1918 — COMMENTED** [review]
There is an issue in the language change and some imports
Apart of the everything works as expected

File: frontend/lib/search/results_loader.dart (hunk 1 above)
Comment (michael-markl) [reply frontend/lib/search/results_loader.dart:11]: Is there no linter for that :o

File: frontend/lib/about/language_change.dart (hunk 4 above)
Comment (michael-markl) [reply frontend/lib/about/language_change.dart:34]: Let's merge #1305 first and check again. I suspect, that the tree recreation removes the snackbar.

File: frontend/lib/search/results_loader.dart (hunk 1 above)
Comment (michael-markl) [reply frontend/lib/search/results_loader.dart]: I went ahead and enabled it and fixed a few more occurences

File: frontend/lib/about/language_change.dart (hunk 4 above)
Comment (f1sh1918) [reply frontend/lib/about/language_change.dart:34]: updated the branch. just some conflicts to resolve

Comment (michael-markl) [reply frontend/lib/about/language_change.dart:34]: Snackbars are visible for me

**Review by f1sh1918 — APPROVED** [review]
Tested on android and ios. works as expected! Also language change is fine now

**Pull request merged**